src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,17 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Machine> {
```

Comment:
The `MachineProducer` interface should be parameterized with `Excavator`, not `Machine`. This ensures that only `Machine` subtypes can be created, preventing the creation of a producer that does not use `Machine` or its subtypes.